platforms/core-runtime/process-services/src/integTest/groovy/org/gradle/process/internal/health/memory/MemoryStatusUpdateIntegrationTest.groovy

```diff
@@ -35,11 +35,13 @@ class MemoryStatusUpdateIntegrationTest extends AbstractIntegrationSpec {
     }
 
     private static String waitForMemoryEventsTask() {
-        return '''
+        return groovyScript('''
             import java.util.concurrent.CountDownLatch
             import org.gradle.process.internal.health.memory.*
 
             task waitForMemoryEvents {
+                def projectDir = project.layout.projectDirectory
+                def logger = logger
```

Comment:
:thought_balloon: I wonder why the `logger` reference is problematic, though. It should be resolved to `Task.getLogger` inside `doLast` clause, but, apparently, it only does so at `logger.warn "Waiting for memory status events..."` and not inside the callbacks, which have `BrokenScript` as `this` and `owner`. I don't fully understand the way Groovy resolves things at this point :(

Reply:
I had similar questions. No clue.

Reply:
Sounds like something we should fix as part of the adoption track: a `logger` reference within a `Closure` given as a `doLast` argument should resolve to `task.logger` without errors.